src/lib/operators/table_scan/column_is_null_table_scan_impl.cpp

```diff
@@ -112,42 +141,116 @@ void ColumnIsNullTableScanImpl::_scan_value_segment(const BaseValueSegment& segm
     return;
   }
 
-  DebugAssert(segment.is_nullable(), "Columns that are not nullable should have been caught by edge case handling.");
-
-  auto iterable = NullValueVectorIterable{segment.null_values()};
+  if constexpr (std::is_same_v<BaseSegmentType, hyrise::BaseValueSegment>) {
+    DebugAssert(segment.is_nullable(), "Columns that are not nullable should have been caught by edge case handling.");
+    _scan_iterable_for_null_values(NullValueVectorIterable{segment.null_values()}, chunk_id, matches, position_filter);
+  } else if constexpr (std::is_same_v<BaseSegmentType, hyrise::BaseDictionarySegment>) {
+    DebugAssert(segment.unique_values_count() != 0 && segment.unique_values_count() != segment.size(),
+                "DictionarySegments without or with exclusivly NULLs should have been caught by edge case handling.");
+    _scan_iterable_for_null_values(create_iterable_from_attribute_vector(segment), chunk_id, matches, position_filter);
+  } else {
+    const auto& null_values = segment.null_values();
+    DebugAssert(null_values.has_value(),
+                "Segment without null_values vector should have been caught by edge case handling.");
+    _scan_iterable_for_null_values(NullValueVectorIterable{*null_values}, chunk_id, matches, position_filter);
+  }
+}
 
-  const auto invert = _predicate_condition == PredicateCondition::IsNotNull;
+template <typename BaseIterableType>
+void ColumnIsNullTableScanImpl::_scan_iterable_for_null_values(
+    const BaseIterableType& iterable, const ChunkID chunk_id, RowIDPosList& matches,
+    const std::shared_ptr<const AbstractPosList>& position_filter) const {
+  const auto invert = predicate_condition == PredicateCondition::IsNotNull;
   const auto functor = [&](const auto& value) {
     return invert ^ value.is_null();
   };
-  iterable.with_iterators([&](auto iter, auto end) {
+
+  iterable.with_iterators(position_filter, [&](auto iter, auto end) {
     _scan_with_iterators<false>(functor, iter, end, chunk_id, matches);
   });
 }
 
+template <>
+bool ColumnIsNullTableScanImpl::_matches_all(const BaseDictionarySegment& segment) const {
+  switch (predicate_condition) {
+    case PredicateCondition::IsNull:
+      return segment.unique_values_count() == 0;
+
+    case PredicateCondition::IsNotNull:
+      return segment.unique_values_count() == segment.size();
```

Comment:
Short comment that we can only be sure if segment contains unique, non-NULL values?